lib/charms/observability_libs/v1/cert_handler.py

```diff
@@ -355,6 +360,22 @@ def __init__(
             self._on_upgrade_charm,
         )
 
+        if refresh_events:
+            for ev in refresh_events:
+                self.framework.observe(ev, self._on_refresh_event)
+
+    def _on_refresh_event(self, _):
+        # Renew only if there are CSR changes
+        curr_csr = self._csr.encode() if self._csr else None
+        new_csr = generate_csr(
```

Comment:
it's not entirely clear to me why we need to generate a new csr in order to see if the one we have already is out of date. What things could possibly have changed? sans-ip, sans-dns, cert_subject, private-key, or all of the above?
I agree that it's easier to simply generate a new csr and compare with the old one instead of putting all these input params in, say, storedstate, and comparing them in order to determine if we need to renew our csr. But, we should be explaining in a comment or docstring why we're choosing this approach.

Reply:
All fields' changes are valid to generate a new csr, even the private key. Added docstrings to explain better.